corehq/apps/app_manager/suite_xml/post_process/instances.py

```diff
@@ -110,18 +110,20 @@ def __call__(self, fn):
         return fn
 
 
-INSTANCE_BY_ID = {
-    'groups': Instance(id='groups', src='jr://fixture/user-groups'),
-    'reports': Instance(id='reports', src='jr://fixture/commcare:reports'),
-    'ledgerdb': Instance(id='ledgerdb', src='jr://instance/ledgerdb'),
-    'casedb': Instance(id='casedb', src='jr://instance/casedb'),
-    'commcaresession': Instance(id='commcaresession', src='jr://instance/session'),
+INSTANCE_KWARGS_BY_ID = {
+    'groups': dict(id='groups', src='jr://fixture/user-groups'),
+    'reports': dict(id='reports', src='jr://fixture/commcare:reports'),
+    'ledgerdb': dict(id='ledgerdb', src='jr://instance/ledgerdb'),
+    'casedb': dict(id='casedb', src='jr://instance/casedb'),
+    'commcaresession': dict(id='commcaresession', src='jr://instance/session'),
 }
 
 
-@register_factory(*INSTANCE_BY_ID.keys())
+@register_factory(*INSTANCE_KWARGS_BY_ID.keys())
 def preset_instances(domain, instance_name):
-    return INSTANCE_BY_ID.get(instance_name, None)
+    kwargs = INSTANCE_KWARGS_BY_ID.get(instance_name, None)
+    if kwargs:
+        return Instance(**kwargs)
```

Comment:
> I haven't dug too deeply into the eulxml.xmlmap source, but I'm assuming that the NodeListField expects the nodes you pass it to not exist elsewhere in the tree

Ah, this makes sense (and also makes sense why you were worried about the uniqueness).

But I still don't quite understand why this wouldn't come up before? I see 37 occurrences of `<instance id="casedb"` in the enikshay suite.

>  I can try to confirm this explanation if you think that's worth spending time on

no that seems fine, though might try and dig in and satisfy my own intellectual curiosity at some point